### ISSUE TITLE
Setup unit testing

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/market#1344]
We try that again with testing, and putting back unit testing.

### Issue goal

Tests run as part of CI, collecting coverage which is sent to CodeClimate so we get coverage reports and status under each PR. Each `@shared/atoms` component folder should include a minimum render test for default component state.

### Options from former codebase

1. use [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/), create basic rendering test every component should go through
2. use [Storybook](https://storybook.js.org) and make stories run as unit tests too, which would cover most components without writing additional unit tests for them, [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/) might still be required additionally

Option 2. might be way more tricky than 1, but would provide us additionally with a nice tool for development.

As test runner most likely [Jest](https://jestjs.io) stays a good choice.

**Comment by EnzoVezzaro** [comment]
I like the second choice better. Using Storybook will allow us to have a nice UI Library to share/test more easily, and we can integrate it with [react-testing-library](https://storybook.js.org/docs/react/writing-tests/importing-stories-in-tests#example-with-testing-library) anyway during setup to have the whole testing capabilities enabled. As test runner, [Jest](https://testing-library.com/docs/react-testing-library/intro) would be the recommended one in this case. For me it's clear that Storybook would be a superior solution. 

@kremalicious I have a question about this part: "collecting coverage which is sent to CodeClimate so we get coverage reports and status under each PR". I'm not really sure how this is handled, do you know how this integration works? can you point me to any documentation about it?

**Comment by kremalicious** [comment]
See ocean.js for coverage setup: 
https://github.com/oceanprotocol/ocean.js/blob/v4main/.github/workflows/ci.yml#L169-L185

Full docs:
https://docs.codeclimate.com/docs/configuring-test-coverage

**Comment by EnzoVezzaro** [comment]
> See ocean.js for coverage setup: https://github.com/oceanprotocol/ocean.js/blob/v4main/.github/workflows/ci.yml#L169-L185
> 
> which we have in here commented out: https://github.com/oceanprotocol/market/blob/v4/.github/workflows/ci.yml#L45-L55
> 
> Full docs: https://docs.codeclimate.com/docs/configuring-test-coverage

great. Thanks ✌🏽

**Comment by EnzoVezzaro** [comment]
How in detph are we willing to go with Storybook? As they have very extensive UI testing tools, see [here](https://storybook.js.org/docs/react/writing-tests/introduction). Probably all we need is [this one](https://storybook.js.org/docs/react/writing-tests/importing-stories-in-tests#example-with-testing-library), but they have other features we might wanna use ([Visual tests](https://storybook.js.org/docs/react/writing-tests/visual-testing), [Accessibility tests](https://storybook.js.org/docs/react/writing-tests/accessibility-testing), [Interaction tests](https://storybook.js.org/docs/react/writing-tests/interaction-testing), [Snapshot tests](https://storybook.js.org/docs/react/writing-tests/snapshot-testing))

**Comment by EnzoVezzaro** [comment]
@kremalicious what do you think is the best place for these stories? I like to have them on the same folder as the component, like so:

- Components > @shared > atoms > Button (for example) > *.stories.tsx, *.tsx, *.module.css

so we can have all component's file in the same directory. This is just a small change to the default Storybook's file structure where you have a `storybook` folder with all stories inside `src`.

What do you think?

**Comment by kremalicious** [comment]
yes, we use component folders so also perfect place to put stories and/or tests right beside the component.
